### PR TITLE
Add Contentful link checker

### PR DESCRIPTION
**Pull request by uoa-noel**
Add Contentful App Framework app for ensuring pages on ResearchHub are only linked to by _one_ SubHub internal pages field.

### CODE REVIEW

File: subhub-link-checker/src/components/Field.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+interface FieldProps {
+  sdk: FieldExtensionSDK;
+}
+
+function checkPageReferences(sdk: FieldExtensionSDK, subhubSys: EntrySys, pageSys: EntrySys): Promise<Boolean> {
+  console.log(`Checking page reference for subhub ${subhubSys.id} and page ${pageSys.id}`);
+  // First, check if the content author is trying to link a subhub to itself.
+  if (subhubSys.type === pageSys.type && subhubSys.id === pageSys.id) {
+    console.log("Subhub and page are the same, rejecting.");
```
Comment (warrickct) [inline subhub-link-checker/src/components/Field.tsx:16]: Console log. You can choose if you want to remove these.

File: subhub-link-checker/src/components/Field.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    return Promise.resolve(false);
+  }
+  // Fetch other subhubs that have links to this page.
+  return sdk.space.getEntries({
+    "content_type": "subHub",
+    "fields.internalPages.sys.id": pageSys.id,
+    "sys.id[ne]": subhubSys.id
+  }).then((entries: CollectionResponse<Object>) => {
+    console.log(`Found ${entries.items.length} other Subhub(s) that contain this page.`);
```
Comment (warrickct) [inline subhub-link-checker/src/components/Field.tsx:25]: Console log. You can choose if you want to remove these.

File: subhub-link-checker/src/components/Field.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    if (entries.items.length > 0) {
+      console.log("Other SubHub(s) containing this page",entries.items);
```
Comment (warrickct) [inline subhub-link-checker/src/components/Field.tsx:27]: Console log. You can choose if you want to remove these.

File: subhub-link-checker/src/components/Field.tsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+          }
+        });
+      }).catch(reason => {
+        sdk.dialogs.openAlert({
+          title: "Error occurred in the SubHub link checking application",
+          message: "Sorry, an error occurred while adding your page(s) to the SubHub Internal Pages field. " +
+            "Please try again. If problems persist, please message the ResearchHub team for assistance. " +
+            "You can also add your page(s) to the External Pages field as a workaround."
+        });
+        console.log("Error occurred in SubHub link checking application", reason);
```
Comment (warrickct) [inline subhub-link-checker/src/components/Field.tsx:109]: Console log. You can choose if you want to remove these.

**Review by warrickct — APPROVED** [review]
Looks good to me. You might want to decide which console logs to remove if any. This was a pretty insightful PR.

File: subhub-link-checker/src/react-app-env.d.ts [hunk 5]
```diff
@@ -0,0 +1 @@
+/// <reference types="react-scripts" />
```
[the remark this answers is not in the document]
Comment (uoa-noel) [reply subhub-link-checker/src/react-app-env.d.ts:1]: This is part of the generated type declarations file that's part of create-contentful-app. :)

**Review by rosemcc — APPROVED** [review]

**Pull request merged**